lib/oauth2/client.rb

```diff
@@ -201,7 +201,7 @@ def assertion
     # @return [Hash] the params to add to a request or URL
     def redirection_params
       if options[:redirect_uri]
-        {'redirect_uri' => options[:redirect_uri]}
+        {'redirect_uri' => options.delete(:redirect_uri)}
```

Comment:
how about....
```ruby
def redirection_params
  options.transform_keys!{ |key| key.to_s }
end
```

Reply:
Hash#transform_keys! is new feature in ruby 2.5.
We need to support [2.3 or higher in v2.x](https://github.com/oauth-xx/oauth2#rubies-with-continued-support-past-oauth2-2x).

Reply:
@mathieujobin @meganemura We are not dropping any supported Rubies for v2.x.  We will continue to support down to Ruby 1.9 and JRuby 1.7.  We will drop Rubies < MRI 2.3 with version 3.0.0, which I expect to follow version 2.0.0 quickly.